src/main/java/dev/morling/onebrc/CalculateAverage_thanhtrinity.java

```diff
@@ -0,0 +1,256 @@
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+
+        // System.out.println("Num Of Processor:" + TOTAL_PROCESSOR);
+        var threads = new ArrayList<Thread>();
+
+        var fileChannel = FileChannel.open(Path.of(FILE), READ);
+        long fullSize = fileChannel.size();
+        // System.out.println("FullSize:" + fullSize);
+
+        var standardChunkSize = fullSize / TOTAL_PROCESSOR;
+        // System.out.println("StandardChunkSize:" + standardChunkSize);
+        var chunkDataList = new ChunkData[TOTAL_PROCESSOR];
+
+        var start = 0L;
+        var end = standardChunkSize;
+        for (int index = 0; index < TOTAL_PROCESSOR; index++) {
+            long newStart = start;
+            end = adjustBreakLinePosition(start + standardChunkSize) + 1;
+            end = end >= fullSize ? fullSize : end;
+            var chunkSize = end - start;
+
+            // Have checked with virtual thread but it slower than normal thread
+            int taskIdx = index;
+            var thread = new Thread(() -> {
+                try (var file = new RandomAccessFile(FILE, "r");
+                        var fc = file.getChannel()) {
+                    var buffer = fc.map(READ_ONLY, newStart, chunkSize);
+                    chunkDataList[taskIdx] = processBufferData(buffer, taskIdx + 1);
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+            });
+            thread.start();
+            threads.add(thread);
+            if (end == fullSize) {
+                break;
+            }
+            start = end;
+        }
+
+        for (var thread : threads) {
+            thread.join();
+        }
+        consolidateData(chunkDataList);
+    }
+
+    private static long adjustBreakLinePosition(long position) throws IOException {
+        try (var file = new RandomAccessFile(FILE, "r")) {
+            file.seek(position);
+            while (file.read() != '\n' && position < file.length()) {
+                position++;
+            }
+        }
+        return position;
+
+    }
+
+    private static ChunkData processBufferData(MappedByteBuffer buffer, long taskIdx) {
+
+        final int citySize = 100000;
+        var breakLineIndex = 0;
+
+        var cities = new City[citySize];
+        City city = null;
+        var isProcessKey = true;
+        var hashKey = 0;
+        double result = 0;
+        int integerPart = 0;
+        double fractionalPart = 0;
+        boolean isFractional = false;
+        double divisorForFraction = 1;
+        boolean isNegative = false;
+        for (int i = 0; i < buffer.limit(); i++) {
+            var b = buffer.get();
+            var position = i + 1;
+            if (isProcessKey) {
+                if (b == ';') {
+                    int cIdx = abs(hashKey % citySize);
+                    city = cities[cIdx];
```

Comment:
How do you deal with hash collisions here? Can't seem to find any sort of linear probing or the like in that case.